pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -23,15 +23,21 @@ import (
 	"go.opentelemetry.io/collector/client"
 	conventions "go.opentelemetry.io/collector/model/semconv/v1.5.0"
 	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/kube"
 )
 
 // extractPodIds extracts IP and pod UID from attributes or request context.
 // It returns a value pair containing configured label and IP Address and/or Pod UID.
 // If empty value in return it means that attributes does not contains configured label to match resources for Pod.
-func extractPodID(ctx context.Context, attrs pcommon.Map, associations []kube.Association) (podIdentifierKey string, podIdentifierValue kube.PodIdentifier) {
-	connectionIP := getConnectionIP(ctx)
+func extractPodID(
```

Comment:
We could also return error and log error outside of this function

Reply:
This would work for the sites where we return immediately, but not for `getConnectionIP`, where we continue even if we don't get anything. Maybe it's just not worth logging in that case?

Reply:
Well, the log in `getConnectionIp` was not mentioned in the issue (that function didn't even exist when the issue was created), so I'll just follow this suggestion and ignore that log.